From ee73c255d75165607c7f04defd5e8edfa3524914 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Thu, 08 Sep 2011 18:53:45 +0000
Subject: window: make sure not to unref NULL GFiles

The GFile pointed by parent here can be NULL after the cycle, so fold
the unref in the != NULL block.
---
diff --git a/src/nautilus-window-manage-views.c b/src/nautilus-window-manage-views.c
index 247f66a..66d47cf 100644
--- a/src/nautilus-window-manage-views.c
+++ b/src/nautilus-window-manage-views.c
@@ -712,11 +712,13 @@ begin_location_change (NautilusWindowSlot *slot,
 			from_folder = parent;
 			parent = g_file_get_parent (from_folder);
 		}
+
 		if (parent != NULL) {
 			new_selection = g_list_prepend (NULL, nautilus_file_get (from_folder));
+			g_object_unref (parent);
 		}
+
 		g_object_unref (from_folder);
-		g_object_unref (parent);
 	}
 
 	end_location_change (slot);
--
cgit v0.9.0.2
